### ISSUE TITLE
awscdk/require-passing-this should not flag resources defined outside Construct class

### ISSUE THREAD

**Issue by JohnnyOu** [ren-yamanashi/eslint-cdk-plugin#427]
### Describe the bug

The rule flags standalone/helper functions that accept a `scope: Construct` parameter and pass it to a new Construct. The rule should only apply inside Construct class constructors/methods where this is available as an alternative. In a utility function, there is no `this` to pass, so the rule should not flag.

### Expected Behavior

Rule should not trigger in standalone utility functions outside Construct class

### eslint-plugin-awscdk Version

4.3.2

### ESLint Version

10.2.1

### Language Version

5.9.3

### Additional Context

Valid code example:
```typescript
class MyStack {
  constructor(parent: App, id: string, props: MyStackProps) {
    // this should be flagged
    new MyConstruct(parent, id, { ...someProps });
  }
}

export function createMyResource(parent: App, id: string, someProps: SomeProps) {
  // this should not be flagged
  new MyConstruct(parent, id, { ...someProps });
}
```

**Comment by ren-yamanashi** [comment]
Thanks for the bug report! I'll fix it.
